Description: revert cp -n behavior to debian 12 and prior
Author: Michael Stone <mstone@debian.org>
Bug-Debian: https://bugs.debian.org/1058752
--- a/src/cp.c
+++ b/src/cp.c
@@ -192,8 +192,9 @@
   -L, --dereference            always follow symbolic links in SOURCE\n\
 "), stdout);
       fputs (_("\
-  -n, --no-clobber             do not overwrite an existing file (overrides a\n\
-                                 -u or previous -i option). See also --update\n\
+  -n, --no-clobber             do not overwrite an existing file and do not fail\n\
+                                 (overrides a -u or previous -i option). See also\n\
+                                 --update; equivalent to --update=none.\n\
 "), stdout);
       fputs (_("\
   -P, --no-dereference         never follow symbolic links in SOURCE\n\
@@ -1067,7 +1068,9 @@
           break;
 
         case 'n':
-          x.interactive = I_ALWAYS_NO;
+          x.interactive = I_ALWAYS_SKIP;
+          x.update = false;
+          error (0, 0, _("warning: behavior of -n is non-portable and may change in future; use --update=none instead"));
           break;
 
         case 'P':
--- a/tests/cp/cp-i.sh
+++ b/tests/cp/cp-i.sh
@@ -19,6 +19,8 @@
 . "${srcdir=.}/tests/init.sh"; path_prepend_ ./src
 print_ver_ cp
 
+skip_ "behavior of -n is non-portable, and Debian behavior differs from upstream"
+
 mkdir -p a b/a/c || framework_failure_
 touch a/c || framework_failure_
 
